src/Features/Core/Portable/Diagnostics/DiagnosticAnalyzerService_UpdateSource.cs

```diff
@@ -92,10 +92,10 @@ ImmutableArray<DiagnosticData> IDiagnosticUpdateSource.GetDiagnostics(Workspace
         {
             if (id != null)
             {
-                return GetSpecificCachedDiagnosticsAsync(workspace, id, includeSuppressedDiagnostics, cancellationToken).WaitAndGetResult(cancellationToken);
+                return GetSpecificCachedDiagnosticsAsync(workspace, id, includeSuppressedDiagnostics, cancellationToken).WaitAndGetResult_CanCallOnBackground(cancellationToken);
```

Comment:
it seems super unfortunate that we have a synchronous diagnostics API that blocks on an async implementation.  This synchronous API is also consumed from async clients.  So this just seems like an unnecessary sync point that messes things up.

Reply:
➡️ I agree. This pull request treats this as deferred technical debt with the understanding that it will soon be surfaced under the VSTHRD002 (avoid problematic synchronous waits) umbrella.